Add responsiveness to stories

alshakerM opened the pull request:
UserStories are now responsiveness.

alshakero commented on public/interactive-examples.mdn.mozilla.net/live-examples/css-examples/flexbox/flex-wrap.css:
```diff
@@ -0,0 +1 @@
+#example-element{background-color:#eee;border:.75em solid;padding:.75em;width:80%;display:flex;}#example-element>div{background-color:rgba(0,0,255,.2);border:3px solid #00f;width:60px;margin:10px}
```
This was added by mistake because of overrides in DevTools.

alshakerM replied:
Copy that.
i'll remove it

alshakero commented on public/www.instagram.com/static/bundles/es6/DesktopStoriesPage.css/f3dbb78f17a0.css:
```diff
@@ -0,0 +1,60 @@
+.yUdUG{background-color:#1a1a1a;height:100%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;overflow:hidden;position:relative;width:100%}.aUIsh,.jUFd-{background:0 0;border:0;cursor:pointer;margin:16px;padding:0;position:absolute;top:0}.aUIsh{right:0}.jUFd-{left:0}
```
Same as above.

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -125,8 +135,24 @@ html {
   color: white;
+  cursor: pointer;
+}
+.story-action + .story-action {
```
Add a comment before this because it's not obvious.
```suggestion
// only add margin between two adjacent items
.story-action + .story-action {
```

alshakerM replied:
Done

alshakero replied:
The margin-left: 5px is clear, I mean the comment above the selector (because the selector is the unclear part).

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -157,18 +176,19 @@ html {
   grid-area: left;
   width: min-content;
-  margin-left: -50px;
+  margin-left: -46px;
```
Seems like a magic number. I'd add a comment (similar to the JS doc we added in the JS file).

alshakerM replied:
On it

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -157,18 +176,19 @@ html {
   grid-area: right;
   width: min-content;
   justify-self: flex-end;
+  margin-right: -46px;
```
+1

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -157,18 +176,19 @@ html {
   z-index: 2;
   position: relative;
-  margin-top: -184px;
+  margin-top: -183px;
```
We fixed this by doing -100% I think?

alshakerM replied:
Nah, that was a different one.
If you put -100% here, it would end up being in the middle of the story

alshakero commented on src/UserStories/UserStories.js:
```diff
@@ -44,6 +75,17 @@ export function UserStories({ userId }) {
+  React.useEffect(() => {
+    function handler() {
+      setDimensions(calculateStoryDimensions());
+      console.log(1)
```
Can go.

alshakerM replied:
Done

alshakero commented on src/UserStories/UserStories.js:
```diff
@@ -113,40 +155,34 @@ export function UserStories({ userId }) {
-            const activeStory = userId === user.user_id;
-            const story = activeStory ? activeUserStory : user.stories[0];
+            const activeUser = userId === user.user_id;
+            const story = activeUser ? activeUserStory : user.stories[0];
```
A little out of scope, but I'd a comment here
```suggestion
            // when the user isn't specified, we go to the first story
            const story = activeUser ? activeUserStory : user.stories[0];
```

alshakerM replied:
Done

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -125,8 +135,24 @@ html {
+.story-body {
+  transition: transform 0.5s ease-out;
+  border-radius: 8px;
+  overflow: hidden;
```
Does the story body still need the overflow hidden since it has its own border-radius.

alshakerM replied:
I'll check.

alshakero replied:
That's correct, but usually, you have to explain why to the reviewer because they mistakenly think that the overflow is not needed, "yes, it does" won't change their mind.

alshakerM replied:
Oh, sure.
We need the overflow here for the border radius.

alshakero reviewed (changes requested):
Love it! Left some minor but important comments.

alshakerM commented:
Removed some unneeded code.
Added some comments to make code more understandable.

alshakerM commented:
Changed the Instagram logo text to match the Real IG and made it a link to HomePage

alshakero commented on src/utils.js:
```diff
@@ -39,15 +39,25 @@ export function readableNumber(num) {
 }
 /**
- * 
+ *
```
The description of the function should be in this line, not in `return`. See the function below.

alshakerM replied:
Done

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -157,18 +174,21 @@ html {
   z-index: 2;
   position: relative;
-  margin-top: -184px;
+  margin-top: -41%;
```
This broke the footer when you have a smaller screen.

alshakerM replied:
On it

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -8,31 +8,36 @@ html {
-  padding-left: 50vw;
-  transition: transform 0.5s ease-in-out;
+  transition: transform 0.2s ease-out;
   height: 100vh;
```
I'd make this 
```suggestion
  min-height: 100vh;
```

To show a vertical scroll when the window is wide and short.

alshakerM replied:
Done

alshakero reviewed (changes requested):
Looks amazing. One important bug.

alshakerM commented:
Changed scrollable stories from height to min-height.
Switched from margin-top -41% to 182px in stories input.
Changed the JSdocs on a function in utlis.
Added a comment to make story actions selector be more understandable.

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -8,31 +8,36 @@ html {
   overflow: hidden;
 }
 .stories-scrollable {
+  transition: transform 0.2s ease-out;
```
This is the cause of the weird dance when you flip to the next story. I'd change it to `0.4s` to make it slower than the scroll effect of `stories-container` (which I'd change to `0.2s`).

alshakerM replied:
Done

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -8,31 +8,36 @@ html {
+  margin: 18px 0;
   border-radius: 8px;
-  margin-left: 50px;
+  transition: all 0.5s ease-out;
```
I mean this, it should be faster than the story resize:
```suggestion
  transition: all 0.2s ease-out;
```

alshakerM replied:
Yupe, much better

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -8,31 +8,36 @@ html {
+  flex: 0 0 auto;
+}
+.stories-container.is-expanded {
+  border-left: 46px solid transparent;
+  border-right: 46px solid transparent;
```
These numbers still need a comment.

alshakerM replied:
Done

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -8,31 +8,36 @@ html {
-  font-family: 'Festive', cursive !important;
   margin: 0;
-  z-index: 2;
+  z-index: 20;
```
Do you need a z-index? I don't think so.

alshakerM replied:
Not since we switched back to scaling both items.

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -181,7 +202,7 @@ html {
 
 .story-reply-input {
-  width: 85%;
+  width: 86%;
```
Magic number. `flex: 1` is cleaner. (you'll just need a gap in the parent).

alshakerM replied:
Yea, much better.
I think the gaping is not needed because the distance between the wall of the story and the input in the direct icon is different, so I put some padding to match the real IG.

alshakero commented on src/utils.js:
```diff
@@ -39,15 +39,25 @@ export function readableNumber(num) {
 }
 /**
- * 
+ * takes any number over 1k with a dot and turns it into a comma  e.g.(without: 1.000, with: 1,000)
```
That's not what it does, it just groups digits. 

instead of `123123123`, it does `123,123,123`.

alshakerM replied:
Oh, okay.
On it

alshakero commented on src/UserStories/UserStories.js:
```diff
@@ -8,21 +8,48 @@ import {
+  if (window.innerWidth < window.innerHeight) {
+    // strangely, real IG connects the height of the story to the width of the screen
+    height = Math.min(window.innerWidth - STORY_VERTICAL_MARGIN * 2, maxHeight);
+    width = height * STORY_ASPECT_RATIO;
+  } else {
+    if (window.innerWidth < TABLET_BREAKPOINT) {
+      height = numberBetween(
+        TABLET_STORY_RATIO * window.innerWidth,
+        500,
+        maxHeight
+      );
+
+      width = height * STORY_ASPECT_RATIO;
+    } else {
+      height = maxHeight;
+      width = height * STORY_ASPECT_RATIO;
+    }
+  }
```
Since the `width` is calculated the same way in every branch of these ifs and elses, I'd do the calculation in once place after the main if.

alshakero commented on src/UserStories/UserStories.js:
```diff
@@ -105,48 +146,48 @@ export function UserStories({ userId }) {
-            const story = activeStory ? activeUserStory : user.stories[0];
+            const activeUser = userId === user.user_id;
+            const story = activeUser ? activeUserStory : user.stories[0];
+            // when the user isn't specified, we go to the first story
```
This comment should be for this line
```
const story = activeUser ? activeUserStory : user.stories[0];
```

alshakerM replied:
Done

alshakero reviewed (changes requested):
The UI looks absolutely spectacular. I not only like it, I genuinely enjoy using it. Such a beauty.

Left some minor but important comments.

alshakero commented on src/UserStories/UserStories.css:
```diff
@@ -8,31 +8,36 @@ html {
+  border-left: 46px solid transparent;
+  border-right: 46px solid transparent;
+}
+.stories-container.is-expanded:hover *.next-prev-buttons {
```
The `*` selector isn't adding any meaning. 

```suggestion
.stories-container.is-expanded:hover .next-prev-buttons {
```

alshakero reviewed (approved):
Pristine now!
